tests/rbd_mirror/test_rbd_mirror_image_features.py

```diff
@@ -0,0 +1,223 @@
+"""Test case covered -
+    CEPH-9520 - When a both Pool and image based mirroring is established,
+    verify if change of any image features in primary site should reflect
+    on remote site upon modification.
+
+    Test Case Flow:
+    1. Create a pool on both clusters.
+    2. Configure mirroring (peer bootstrap) between two clusters.
+    3. Enable pool mode journal based mirroring on the pool respectively.
+    4. Verify if change of any image features in primary site gets mirrored in secondary site.
+    5. Enable image mode snapshot based mirroring on the pool respectively.
+    6. Verify if change of any image features in primary site gets mirrored in secondary site.
+"""
+import time
+
+from tests.rbd.rbd_utils import Rbd
+from tests.rbd_mirror.rbd_mirror_utils import rbd_mirror_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_journal_image_feature(rbd_mirror, pool_type, **kw):
+    """Method to validate change of image feature in primary
+    got reflected onto secondary in Journal based RBD mirroring.
+    Args:
+        rbd_mirror: Object
+        pool_type: Replicated or EC pool
+        **kw: test data
+        Example::
+            config:
+                imagesize: 2G
+    Returns:
+        int: The return value. 0 for success, 1 for failure
+    """
+    log.info(
+        "Running RBD journal based mirroring test case for image features reflection"
+    )
+    config = kw.get("config")
+    mirror1 = rbd_mirror.get("mirror1")
+    mirror2 = rbd_mirror.get("mirror2")
+    rbd1, rbd2 = [
+        Rbd(**kw, req_cname=cluster_name)
+        for cluster_name in kw.get("ceph_cluster_dict").keys()
+    ]
+    poolname = config[pool_type]["pool"]
+    imagename = config[pool_type]["image"]
+    imagespec = poolname + "/" + imagename
+    image_feature = "object-map,fast-diff"
+    features = list(image_feature.split(","))
+    # Disable image features in primary cluster
+    mirror1.image_feature_disable(imagespec=imagespec, image_feature=image_feature)
+    # wait till image gets reflected to secondary
+    timeout = 300
+    found = 0
+    while timeout:
+        # Verify image feature reflection in secondary cluster
+        out = rbd2.image_info(poolname, imagename)
+        for word in features:
+            if word not in out["features"]:
+                log.info(f"Image feature {word} got disabled successfully in secondary")
+                found = 1
```

Comment:
return 1 at this point instead of found flag.

Reply:
Removed the `found` flag as suggested by you, it's done.